File: launchschool_exercises/210_small_problems/strings_and_text_processing/capitalize-words.js
/*
  Write a function that takes a string as an argument, and returns that string with the first character of every word capitalized and all subsequent characters in lowercase.

  You may assume that a word is any sequence of non-whitespace characters.
*/

function wordCap(str) {
  return str.split(' ').map(function (word) {
    if (/[a-zA-Z]/.test(word[0])) {
      return word[0].toUpperCase() + word.slice(1, word.length);
    } else {
      return word
    }
  }).join(' ');
}

console.log(wordCap('four score and seven'));       // "Four Score And Seven"
console.log(wordCap('the javaScript language'));    // "The Javascript Language"
console.log(wordCap('this is a "quoted" word'));    // 'This Is A "quoted" Word'

File: launchschool_exercises/109_small_problems/easy7/capitalize_words.rb
def word_cap(str)
  str.split(' ').map do |word|
    if word[0] =~ /[a-z]/
      word.capitalize
    else
      word
    end
  end.join(' ')
end

p word_cap('four score and seven') # == 'Four Score And Seven'
p word_cap('the javaScript language') # == 'The Javascript Language'
p word_cap('this is a "quoted" word') # == 'This Is A "quoted" Word'
